crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -54,7 +52,12 @@ impl BatchBuilder {
     /// A pool of batch-proving workers is spawned, which are fed new batch jobs periodically.
     /// A batch is skipped if there are no available workers, or if there are no transactions
     /// available to batch.
-    pub async fn run(self, mempool: SharedMempool, store: StoreClient) {
+    pub async fn run(
+        self,
+        mempool: SharedMempool,
+        store: StoreClient,
+        batch_prover: BatchProver,
+    ) {
```

Comment:
Related to the previous comment, if we make `batch_prover` a field of `BatchBuilder`, we won't need to pass it here.

Also, not from this PR, but we have a slight inconsistency here between batch and block builders: `BlockBuilder` takes the store as a construction parameter, but `BatchBuilder` takes it via the `run` method. Should we also move store into the constructor for `BatchBuilder`? @Mirko-von-Leipzig - what do you think?

Reply:
I've been meaning to relook at these as part of #676.

Its also semi-related to properly decoupling components, i.e. a block-producer should be able to startup independently from the store.

I think we can align them now; but we should take a closer look in the medium term. Bit of a non-answer I know 🙃